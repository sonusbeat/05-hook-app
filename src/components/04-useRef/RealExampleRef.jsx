import { useState } from "react";
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <section>
      <h2 className="text-center">Real Example Ref</h2>

      <hr />

      { show && <MultipleCustomHooks /> }

      <div className="form-group text-center mt-5">
        <button
          className="btn btn-primary"
          onClick={() => setShow(!show)}
        >Mostrar / Ocultar</button>
      </div>
    </section>
  );
};

export default RealExampleRef;
