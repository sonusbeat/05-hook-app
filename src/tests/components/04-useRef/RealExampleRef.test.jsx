import RealExampleRef from '../../../components/04-useRef/RealExampleRef';
import { shallow } from 'enzyme';

describe('Pruebas en <RealExampleRef />', () => {

  const wrapper = shallow(<RealExampleRef />);

  test('DeberÃ­a de mostrarse correctamente', () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);

  });

  test('Debe de mostrar el componente <MultipleCustomHooks />', () => {

    wrapper.find("#toggle").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);

  });

});
