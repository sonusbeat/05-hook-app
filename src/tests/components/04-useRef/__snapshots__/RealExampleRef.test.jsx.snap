// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> Deber√≠a de mostrarse correctamente 1`] = `
<section>
  <h2
    className="text-center"
  >
    Real Example Ref
  </h2>
  <hr />
  <div
    className="form-group text-center mt-5"
  >
    <button
      className="btn btn-primary"
      id="toggle"
      onClick={[Function]}
    >
      Mostrar / Ocultar
    </button>
  </div>
</section>
`;
